src/test_util/mod.rs

```diff
@@ -348,6 +348,64 @@ where
     .await
 }
 
+// Retries a func every `retry` duration until given an Ok(T); panics after `until` elapses
+pub async fn retry_until<F, Fut, T, E>(mut f: F, retry: Duration, until: Duration) -> T
+where
+    F: FnMut() -> Fut,
+    Fut: Future<Output = Result<T, E>> + Send + 'static,
+{
+    let started = Instant::now();
+    while started.elapsed() < until {
+        match f().await {
+            Ok(res) => return res,
+            Err(_) => tokio::time::delay_for(retry).await,
+        }
+    }
+    panic!("Timeout")
+}
```

Comment:
Would it possible to replace `wait_for_duration` by `retry_until`?

Reply:
It should be possible to replace most instances of `wait_for_tcp` and `wait_for_tcp_duration`, but I didn't want to make too many changes yet. Can follow-up in another PR if this one is approved.

Reply:
Sounds good :+1: